x/concentrated-liquidity/position.go

```diff
@@ -118,26 +140,60 @@ func (k Keeper) setPosition(ctx sdk.Context,
 	positionId uint64,
 ) {
 	store := ctx.KVStore(k.storeKey)
-	key := types.KeyFullPosition(poolId, owner, lowerTick, upperTick, joinTime, freezeDuration, positionId)
-	osmoutils.MustSetDec(store, key, liquidity)
+
+	// Create a new Position object with the provided information.
+	position := model.Position{
+		PositionId:     positionId,
+		PoolId:         poolId,
+		Address:        owner.String(),
+		LowerTick:      lowerTick,
+		UpperTick:      upperTick,
+		JoinTime:       joinTime,
+		FreezeDuration: freezeDuration,
+		Liquidity:      liquidity,
+	}
+
+	// Set the position ID to position mapping.
+	key := types.KeyPositionId(positionId)
+	osmoutils.MustSet(store, key, &position)
+
+	// Set the address-pool-position ID to position mapping.
+	key = types.KeyAddressPoolIdPositionId(owner, poolId, positionId)
+	store.Set(key, sdk.Uint64ToBigEndian(positionId))
+
+	// Set the pool ID to position ID mapping.
+	key = types.KeyPoolPositionPositionId(poolId, positionId)
+	store.Set(key, sdk.Uint64ToBigEndian(positionId))
 }
 
 func (k Keeper) deletePosition(ctx sdk.Context,
-	poolId uint64,
-	owner sdk.AccAddress,
-	lowerTick, upperTick int64,
-	joinTime time.Time,
-	freezeDuration time.Duration,
 	positionId uint64,
+	owner sdk.AccAddress,
+	poolId uint64,
 ) error {
 	store := ctx.KVStore(k.storeKey)
-	key := types.KeyFullPosition(poolId, owner, lowerTick, upperTick, joinTime, freezeDuration, positionId)
 
+	// Remove the position ID to position mapping.
+	key := types.KeyPositionId(positionId)
+	if !store.Has(key) {
+		return types.PositionIdNotFoundError{PositionId: positionId}
+	}
+	store.Delete(key)
+
+	// Remove the address-pool-position ID to position mapping.
+	key = types.KeyAddressPoolIdPositionId(owner, poolId, positionId)
 	if !store.Has(key) {
-		return types.PositionNotFoundError{PoolId: poolId, LowerTick: lowerTick, UpperTick: upperTick, JoinTime: joinTime, FreezeDuration: freezeDuration}
+		return fmt.Errorf("position id %d not found for address %s and pool id %d", positionId, owner.String(), poolId)
```

Comment:
nit: consider defining an error in `types`

Reply:
Defined here https://github.com/osmosis-labs/osmosis/pull/4695/commits/4c4e12b4cf847ed7cf51f02cac81d8b790c97589